subprojects/signing/src/main/java/org/gradle/plugins/signing/signatory/Signatory.java

```diff
@@ -45,4 +45,13 @@
      * @return The raw bytes of the signature
      */
     byte[] sign(InputStream toSign);
+
+    /**
+     * Returns an object that represents the value of the {@code signatory} property in the Sign task. The value
+     * returned by this method should change when the configuration of the signatory changes in a way that would
+     * change the output of the Sign task.
+     *
+     * @return An object representing the value of the {@code signatory} property.
+     */
+    Object getInputProperty();
```

Comment:
Input property sounds very generic. We should find a better name.

Reply:
Good point. I've renamed the method to `getKeyId()` and it now returns a 'String` (see 
56dc553)